internal/ethapi/api.go

```diff
@@ -68,8 +68,13 @@ func (s *PublicEthereumAPI) GasPrice(ctx context.Context) (*hexutil.Big, error)
 	if err != nil {
 		return nil, err
 	}
-	if head := s.b.CurrentHeader(); head.BaseFee != nil {
-		tipcap.Add(tipcap, head.BaseFee)
+	bNrOrHash := rpc.BlockNumberOrHashWithNumber(rpc.PendingBlockNumber)
```

Comment:
Why not using `s.b.PendingBlockAndReceipts()`?

Reply:
yeah, make sense. changed to use s.b.PendingBlockAndReceipts().